apps/api/src/app/environments-v1/usecases/output-renderers/email-output-renderer.usecase.ts

```diff
@@ -43,6 +44,30 @@ export class EmailOutputRendererUsecase {
     return { subject: subject as string, body: renderedHtml };
   }
 
+  private removeTrailingEmptyLines(node: TipTapNode): TipTapNode {
+    if (!node.content || node.content.length === 0) return node;
+
+    // Iterate from the end of the content and find the first non-empty node
+    let lastIndex = node.content.length;
+    // eslint-disable-next-line no-plusplus
+    for (let i = node.content.length - 1; i >= 0; i--) {
+      const childNode = node.content[i];
+
+      const isEmptyParagraph =
+        childNode.type === 'paragraph' && !childNode.text && (!childNode.content || childNode.content.length === 0);
```

Comment:
question: what about the other types like heading or text, don't they create an issue?

Reply:
Good catch, they might cause the issue as well.
i will check and update accordingly Thanks 🙌🏼

Reply:
The other content types don't cause this issue, as the text or content types properly handle it. However, when someone leaves an empty bullet/unordered/numbered list at the end, it leads to clipping. 
I didn't handle that case as it will increase the complexity of this code and also an empty bullet list is easily identifiable on the email editor and will be very unlikely to miss it